hedera-node/src/main/java/com/hedera/services/txns/token/TokenUpdateTransitionLogic.java

```diff
@@ -65,125 +67,195 @@
 	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
 
 	private final boolean allowChangedTreasuryToOwnNfts;
-	private final TokenStore store;
+	private final TransactionContext txnCtx;
 	private final HederaLedger ledger;
 	private final OptionValidator validator;
-	private final TransactionContext txnCtx;
 	private final Predicate<TokenUpdateTransactionBody> affectsExpiryOnly;
+	private final TypedTokenStore tokenStore;
+	private final AccountStore accountStore;
 
 	public TokenUpdateTransitionLogic(
 			boolean allowChangedTreasuryToOwnNfts,
-			OptionValidator validator,
-			TokenStore store,
 			HederaLedger ledger,
+			OptionValidator validator,
 			TransactionContext txnCtx,
-			Predicate<TokenUpdateTransactionBody> affectsExpiryOnly
+			Predicate<TokenUpdateTransactionBody> affectsExpiryOnly,
+			TypedTokenStore tokenStore,
+			AccountStore accountStore
 	) {
 		this.validator = validator;
-		this.store = store;
-		this.ledger = ledger;
 		this.txnCtx = txnCtx;
 		this.affectsExpiryOnly = affectsExpiryOnly;
 		this.allowChangedTreasuryToOwnNfts = allowChangedTreasuryToOwnNfts;
+		this.ledger = ledger;
+		this.tokenStore = tokenStore;
+		this.accountStore = accountStore;
 	}
 
 	@Override
 	public void doStateTransition() {
-		try {
-			transitionFor(txnCtx.accessor().getTxn().getTokenUpdate());
-		} catch (Exception e) {
-			log.warn("Unhandled error while processing :: {}!", txnCtx.accessor().getSignedTxnWrapper(), e);
-			abortWith(FAIL_INVALID);
+		final var op = txnCtx.accessor().getTxn().getTokenUpdate();
+		final var token = tokenStore.loadToken(Id.fromGrpcToken(op.getToken()));
+
+		throwIfExpirationIsInvalid(op);
+		throwIfTokenIsImmutable(op, token);
```

Comment:
We are executing this validation in the `token.update` method. Why do we need to do it here as-well? Generally, validations in the transition logic are "sanity" checks executed only against the `grpc operation`.